packages/flutter/lib/src/material/search.dart

```diff
@@ -89,6 +89,19 @@ Future<T> showSearch<T>({
 /// for another [showSearch] call.
 abstract class SearchDelegate<T> {
 
+  /// The text input action configuring the soft keyboard to a particular action
+  /// button.
+  ///
+  /// Defaults to [TextInputAction.search].
+  final TextInputAction textInputAction = TextInputAction.search;
+
+  /// The hint text to be displayed in the search bar, prompting the user to
+  /// enter an input.
+  ///
+  /// Defaults to [MaterialLocalizations.searchFieldLabel].
+  String getHintText(BuildContext context) =>
```

Comment:
What was your motivation for making this a function?  The alternative would be to pass it in as a String and use the localization value if it's null.  Same thing either way but I'm wondering if you have any specific use for it being a function.

Reply:
Either works, but I do agree, your alternative is much cleaner. Fixing!